src/NServiceBus.Core/Unicast/Transport/TransactionSettings.cs

```diff
@@ -28,6 +28,7 @@ internal TransactionSettings(ReadOnlySettings settings)
         /// <param name="doNotWrapHandlersExecutionInATransactionScope">Should handlers be wrapped</param>
         public TransactionSettings(bool isTransactional, TimeSpan transactionTimeout, IsolationLevel isolationLevel, bool suppressDistributedTransactions, bool doNotWrapHandlersExecutionInATransactionScope)
         {
+            Guard.AgainstNegative(transactionTimeout, "transactionTimeout");
```

Comment:
I wonder if we should also validate the timeout passed against the machine setting ?